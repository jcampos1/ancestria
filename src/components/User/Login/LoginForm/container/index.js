import { connect } from 'react-redux';
import LoginForm from 'components/User/Login/LoginForm/index';
import {setAuthenticated} from 'reducers/User/actions/index';

const mapStateToProps = state => ({
}); 

const mapDispatchToProps = dispatch => ({
	setAuthenticated: () => { dispatch(setAuthenticated()) }
});

const LoginFormContainer = connect(mapStateToProps, mapDispatchToProps)(LoginForm);
export default LoginFormContainer;