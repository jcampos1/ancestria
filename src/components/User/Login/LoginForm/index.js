import React from 'react';
import { Field, reduxForm } from 'redux-form';
import ButtonForm from 'components/common/ButtonForm/index';

export class LoginForm extends React.Component {
	render() {
		const {handleSubmit, isLoading, errors} = this.props;

		return (
			<form name="login-form" onSubmit={handleSubmit}>
				{
					errors.length > 0 &&
						<div className="alert alert-danger" role="alert">
		  					{
		  						errors.map((error, index) => (
		  							<label key={`error_login{index}`}>{error}</label>
		  						))
		  					}
						</div>
				}
				<div>
					<label htmlFor="username">Username or email</label>
					<Field name="username" component="input" type="text" className="form-control" />
				</div>
				<div>
					<label htmlFor="password">Password</label>
					<Field name="password" component="input" type="password" className="form-control" />
				</div>
				<ButtonForm 
					name="Log in" 
					nameLoading="Logging in"
					isLoading={isLoading}/>
			</form>
		) 
	}
}

const LoginReduxForm = reduxForm({
  // a unique name for the form
  form: 'loginForm'
})(LoginForm)

export default LoginReduxForm;



