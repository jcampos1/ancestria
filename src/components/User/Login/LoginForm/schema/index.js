import * as yup from 'yup';
import i18n from 'i18n';
import format from 'string-format';

const validationSchema = () => 
	yup.object().shape({
	  username: yup
	      .string()
	      .max(255)
	      .required(i18n.t('validations.required')),
	  password: yup
	      .string()
	      .min(8, format(i18n.t('validations.min'), '8'))
	      .required(i18n.t('validations.required'))
	});

export default validationSchema;