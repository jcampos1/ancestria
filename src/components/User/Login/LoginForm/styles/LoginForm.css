.login-form h2.login-form__title {
  font-weight: 700;
  color: #83e2ff !important; }

.login-form .login-form__tab {
  padding-left: 50px; }
  .login-form .login-form__tab .login-form__tab__create_account {
    color: #4172d0 !important;
    font-weight: 700; }
  .login-form .login-form__tab a:link {
    text-decoration: none; }

.login-form .login-form__small label {
  display: table-cell;
  vertical-align: middle;
  position: relative; }

.login-form.rounded {
  border-radius: 20px !important; }

@media (max-width: 576px) {
  .login-form h2.login-form__title {
    font-size: 22px; }
  .login-form .login-form__tab {
    padding-left: 0; }
    .login-form .login-form__tab label:first-child {
      padding-left: 50px; } }
