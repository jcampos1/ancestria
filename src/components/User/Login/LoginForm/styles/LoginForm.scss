.login-form {
	h2.login-form__title {
    	font-weight: 700;
    	color: #83e2ff !important;
	}

	.login-form__tab {
		padding-left: 50px; 
		.login-form__tab__create_account {
			color: #4172d0 !important;
    		font-weight: 700;
		}
		a:link {
			text-decoration: none;
		}
	}

	.login-form__small {
		label {
		    display: table-cell;
    		vertical-align: middle;
    		position: relative;
		}
	}

	&.rounded {
		border-radius: 20px !important;
	}
}

@media (max-width: 576px) {
	.login-form {
		h2.login-form__title {
			font-size: 22px;
		}

		.login-form__tab {
			padding-left: 0; 
			label:first-child {
				padding-left: 50px; 
			}
		}
	}
}