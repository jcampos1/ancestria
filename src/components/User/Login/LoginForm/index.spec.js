import React from 'react';
import {shallow} from 'enzyme';
import {LoginForm} from 'components/User/Login/LoginForm/index';
import sinon from 'sinon';

describe('LoginForm component', () => {
	it('Should render LoginForm component when there are errors', () => {
		const handleSubmit = sinon.spy();
		const isLoading = false;
		const errors = ["Unable to log in with provided credentials."]; 

		const wrapper = shallow(
			<LoginForm 
				handleSubmit={handleSubmit} 
				isLoading={isLoading}
				errors={errors} />);

		expect(wrapper).toMatchSnapshot();

		// expect(wrapper.find("form.login-form div.alert label")).toHaveLength(errors.length);
	});
});