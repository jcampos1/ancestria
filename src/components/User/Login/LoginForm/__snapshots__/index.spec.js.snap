// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component Should render LoginForm component when there are errors 1`] = `
<form
  name="login-form"
  onSubmit={[Function]}
>
  <div
    className="alert alert-danger"
    role="alert"
  >
    <label
      key="error_login{index}"
    >
      Unable to log in with provided credentials.
    </label>
  </div>
  <div>
    <label
      htmlFor="username"
    >
      Username or email
    </label>
    <Field
      className="form-control"
      component="input"
      name="username"
      type="text"
    />
  </div>
  <div>
    <label
      htmlFor="password"
    >
      Password
    </label>
    <Field
      className="form-control"
      component="input"
      name="password"
      type="password"
    />
  </div>
  <ButtonForm
    isLoading={false}
    name="Log in"
    nameLoading="Logging in"
  />
</form>
`;
